solidity/src/Base7683.sol

```diff
@@ -87,330 +128,391 @@ abstract contract Base7683 is IOriginSettler, IDestinationSettler {
+        _wordPos = uint248(_nonce >> 8);
+        _bitPos = uint8(_nonce);
     }
 
-    function _resolvedOrder(
-        bytes32 _orderType,
-        address _sender,
-        bytes memory _orderData,
-        uint32 _openDeadline,
-        uint32 _fillDeadline
-    )
-        internal
-        view
-        returns (ResolvedCrossChainOrder memory resolvedOrder, OrderData memory orderData)
-    {
-        if (_orderType != OrderEncoder.orderDataType()) revert InvalidOrderType(_orderType);
-
-        orderData = OrderEncoder.decode(_orderData);
-
-        if (orderData.originDomain != _localDomain()) revert InvalidOriginDomain(orderData.originDomain);
-        if (orderData.sender != TypeCasts.addressToBytes32(_sender)) revert InvalidSender();
-        if (orderData.senderNonce != senderNonce[_sender]) revert InvalidSenderNonce();
-        bytes32 destinationSettler = _mustHaveRemoteCounterpart(orderData.destinationDomain);
-
-        // enforce fillDeadline into orderData
-        orderData.fillDeadline = _fillDeadline;
-
-        // this can be used by the filler to approve the tokens to be spent on destination
-        Output[] memory maxSpent = new Output[](1);
-        maxSpent[0] = Output({
-            token: orderData.outputToken,
-            amount: orderData.amountOut,
-            recipient: destinationSettler,
-            chainId: orderData.destinationDomain
-        });
-
-        // this can be used by the filler know how much it can expect to receive
-        Output[] memory minReceived = new Output[](1);
-        minReceived[0] = Output({
-            token: orderData.inputToken,
-            amount: orderData.amountIn,
-            recipient: bytes32(0),
-            chainId: orderData.originDomain
-        });
-
-        // this can be user by the filler to know how to fill the order
-        FillInstruction[] memory fillInstructions = new FillInstruction[](1);
-        fillInstructions[0] = FillInstruction({
-            destinationChainId: orderData.destinationDomain,
-            destinationSettler: destinationSettler,
-            originData: OrderEncoder.encode(orderData)
-        });
-
-        resolvedOrder = ResolvedCrossChainOrder({
-            user: _sender,
-            originChainId: _localDomain(),
-            openDeadline: _openDeadline,
-            fillDeadline: _fillDeadline,
-            minReceived: minReceived,
-            maxSpent: maxSpent,
-            fillInstructions: fillInstructions
-        });
+    /**
+     * @notice Marks a nonce as used by setting its bit in the appropriate bitmap.
+     * @dev Ensures that a nonce cannot be reused by flipping the corresponding bit in the bitmap.
+     * Reverts if the nonce is already used.
+     * @param _from The address for which the nonce is being used.
+     * @param _nonce The nonce to mark as used.
+     */
+    function _useNonce(address _from, uint256 _nonce) internal {
```

Comment:
IIUC, this is a gas optimization? I would prefer to keep it simple, and if folks want to they can use a child of Base7683 that optimizes it

Reply:
I thought you were talking about un-ordered nonces, I think I took this implementation from Permit2.
I think `mapping(address => mapping(unit256 => bool)) usedNonces` could work as well.